documentation/proposals/Proposal - Windowing.md

```diff
@@ -45,59 +54,70 @@ public interface IWindow : IWindowProperties, IWindowFunctions, IWindowEvents
-	/// </summary>
-	double FramesPerSecond { get; }
+    /// <summary>
+    /// If true, both updates and rendering will happen on the same thread. If false, both updating
+    /// and rendering will be run on their own threads. Default is true.
+    /// </summary>
+    bool UseSingleThreadedWindow { get; }
 
-	/// <summary>
-	/// The number of update operations to run per second. If 0, the update rate will be uncapped. Default is 0.
-	/// </summary>
-	double UpdatesPerSecond { get; }
+    /// <summary>
+    /// The position of the window. Integer vector. If set to -1, use the backend default.
+    /// Default is -1 for both components.
+    /// </summary>
+    Point Position { get; set; }
 
-	/// <summary>
-	/// The graphics API to use, and associated configurations. Default is OpenGL 3.3 forward-compatable core profile.
-	/// </summary>
-	GraphicsAPI API { get; }
+    /// <summary>
+    /// The size of the window. Integer vector. Default is 1280 x 720.
+    /// </summary>
+    Size Size { get; set; }
 
-	/// <summary>
-	/// The title of the window. Default is "Silk.NET Window".
-	/// </summary>
-	string Title { get; set; }
+    /// <summary>
+    /// The number of rendering operations to run per second. If 0, the framerate will be uncapped.
+    /// Default is 0.
+    /// </summary>
+    double FramesPerSecond { get; }
 
-	/// <summary>
-	/// The state of the window. Default is Normal.
-	/// </summary>
-	WindowState WindowState { get; set; }
+    /// <summary>
+    /// The number of update operations to run per second. If 0, the update rate will be uncapped.
+    /// Default is 0.
+    /// </summary>
+    double UpdatesPerSecond { get; }
 
-	/// <summary>
-	/// The type of border the window has. Default is Resizable.
-	/// </summary>
-	WindowBorder WindowBorder { get; set; }
+    /// <summary>
+    /// The graphics API to use, and associated configurations.
+    /// Default is OpenGL 3.3 forward-compatable core profile.
+    /// </summary>
+    GraphicsAPI API { get; }
+    
+    /// <summary>
+    /// The title of the window. Default is "Silk.NET Window".
+    /// </summary>
+    string Title { get; set; }
+    
+    /// <summary>
+    /// The state of the window. Default is Normal.
+    /// </summary>
+    WindowState WindowState { get; set; }
 
-	/// <summary>
-	/// The vertical synchronization mode to use. Default is On.
-	/// </summary>
-	VSyncMode VSync { get; set; }
+    /// <summary>
+    /// The type of border the window has. Default is Resizable.
+    /// </summary>
+    WindowBorder WindowBorder { get; set; }
+    
+    /// <summary>
+    /// The vertical synchronization mode to use. Default is On.
+    /// </summary>
+    VSyncMode VSync { get; set; }
+	
+    /// <summary>
+    /// The number of frames the window needs to be running slowly for before
+    /// <see cref="IWindow.IsRunningSlowly"/> is set to true. Default is 5.
+    /// </summary>
+    int IsRunningSlowlyThreshold { get; set; }
```

Comment:
Change this property name to `SlowFrameTolerance`, or some other names along those lines (put them in the discord if you have suggestions!)